### ISSUE TITLE
core: some trackers are overwriting categories

### ISSUE THREAD

**Issue by ngosang** [Jackett/Jackett#8024]
Some trackers like [XSpeeds](https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/XSpeeds.cs) , TorrentLeech, TorrentDay... are defining some default categories and then overwriting them.

https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/XSpeeds.cs#L40

I think this is error prone and we should remove and avoid using `TorznabUtil.CreateDefaultTorznabTVCaps()` when possible.

@cadatoiva

**Comment by cadatoiva** [comment]
Yes, I agree that we shouldn't be using this function and it should be removed from the code base entirely in favor of an explicit object initializer in the constructor.

I'm not sure I understand the nature of the bug you're describing though? This function lives here, and adds a couple TV categories, but doesn't change or overwrite anything.

https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Utils/TorznabCapsUtil.cs#L11

**Comment by ngosang** [comment]
Here we are adding some TV categories => https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/XSpeeds.cs#L40
And here we are adding the same categories => https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/XSpeeds.cs#L89

**Comment by garfield69** [comment]
Not sure I understand your point.

TorznabUtil.CreateDefaultTorznabTVCaps() adds tv(5000) tvsd(5030) tvhd(5040)  to the table,
While AddCategoryMapping(47, TorznabCatType.TVHD, "TV-HD") adds custom category 100047 to the table, and if necessary adds 5030 too.

**Comment by cadatoiva** [comment]
Yeah, the code doesn't "overwrite" anything, but it is redundant and makes implementation messy.

**Comment by ngosang** [comment]
Example 1 TorrentBytes => https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/TorrentBytes.cs

`TorznabUtil.CreateDefaultTorznabTVCaps():` is adding the category `TorznabCatType.TV` which is not supported by the tracker. **That happens in most trackers and we should think about that..**
If you try to search by that category you will get 0 results.


Example 2, TVVault => https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/TVVault.cs

From TorznabUtil.CreateDefaultTorznabTVCaps():
* TorznabCatType.TV,
* TorznabCatType.TVSD,
* TorznabCatType.TVHD

From category mapping:
* AddCategoryMapping(1, TorznabCatType.TV);
* AddCategoryMapping(2, TorznabCatType.Movies);

We are not using any of them in the search (possibly because the site doesn't support it) and in the response we are trying to guess the TV category with `TvCategoryParser.ParseTvShowQuality(description)`

1. The selected categories don't work because they are not used at all. (we should filter the results at least)
2. We are guessing only tv categories, what happens with movies? Why we show the movies category to the user if he can't search by movies and he won't see movies in the results?

To summarize, I think `TorznabUtil.CreateDefaultTorznabTVCaps()` is error prone and we should use `AddCategoryMapping` when it's possible.

**Comment by ngosang** [comment]
This was fixed in #9831
